logging.js

```diff
@@ -157,17 +157,22 @@ export class LogBuilder {
 }
 
 export class LoggingClient {
-	constructor(appId, logger) {
+	constructor(appId, shouldThrottle, logger) {
 		this._appId = appId;
+		this._shouldThrottle = !!shouldThrottle;
 		this._logger = logger;
+
+		this._uniqueLogs = new Map();
 	}
 
 	log(developerMessage) {
 		this.logBatch([developerMessage]);
 	}
 
 	logBatch(developerMessages) {
-		const logs = developerMessages.map(developerMessage => new LogBuilder(this._appId)
```

Comment:
I figured doing it here in the client made more sense than in the server.  I have to call the function in three places instead of one, but it cuts off the code execution so much sooner.  Otherwise each log will go through all the building and formatting and grabbing extra info for us to compare that we know will be identical.

Reply:
Alternately could do it in a similar format to before by just filtering the existing array:

```javascript
const logs = developerMessages
    .filter(this._throttle)
    .map(developerMessage => new LogBuilder(this._appId)
```